shared/src/price_estimate.rs

```diff
@@ -403,12 +430,42 @@ mod tests {
         let token_a = H160::from_low_u64_be(1);
         let token_b = H160::from_low_u64_be(2);
         let pool_fetcher = Box::new(FakePoolFetcher(vec![]));
-        let estimator = UniswapPriceEstimator::new(pool_fetcher, hashset!());
+        let estimator = UniswapPriceEstimator::new(pool_fetcher, hashset!(), hashset!());
+
+        assert!(estimator
+            .estimate_price(token_a, token_b, 1.into(), OrderKind::Buy)
+            .await
+            .is_err());
+    }
+
+    #[tokio::test]
+    async fn return_error_if_token_denied() {
+        let token_a = H160::from_low_u64_be(1);
+        let token_b = H160::from_low_u64_be(2);
+        let pool_ab = Pool::uniswap(
+            TokenPair::new(token_a, token_b).unwrap(),
+            (10u128.pow(30), 10u128.pow(29)),
+        );
+        let pool_fetcher = Box::new(FakePoolFetcher(vec![pool_ab]));
+        let estimator = UniswapPriceEstimator::new(pool_fetcher, hashset!(), hashset!(token_a));
 
         assert!(estimator
             .estimate_price(token_a, token_b, 1.into(), OrderKind::Buy)
             .await
             .is_err());
+
+        assert!(estimator
+            .estimate_price(token_b, token_a, 1.into(), OrderKind::Buy)
+            .await
+            .is_err());
+
+        // // TODO - make more concise assertion about which error we are getting.
+        // let expected = anyhow!("Buy Token {} Denied!", token_a);
+        // let result = estimator
+        //     .estimate_price(token_b, token_a, 1.into(), OrderKind::Buy)
+        //     .await.unwrap_err();
+        // // Unfortunately we can't compare anyhow errors.
+        // assert_eq!(expected, result);
```

Comment:
The correct way to do it is to use an error enum instead of anyhow. https://docs.rs/thiserror/1.0.24/thiserror/ is nice for that. `anyhow` is meant to be used precisely when you do not need to distinguish errors which is the case for most of our code base.
On the other hand since the distinction is only useful for the test I'm fine with keeping it an anyhow::Error. You could inspect the error string directly through `err.to_string()`.